### PR TITLE
feat(elements): Add & refactor inspectors

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

- Adds Stately's Browser Inspector in development builds for easier tracing/debugging.
  - `npm run app:dev:debug:ui` (from `packages/elements`) or using
  - `NEXT_PUBLIC_CLERK_ELEMENTS_DEBUG_UI=true`
- Moves `@statelyai/inspect` from `dependencies` to `devDependencies`.
- Removes all inspector-related code from the production builds, slimming down the builds.

Fixes SDK-1763
Fixes SDK-1764

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [x] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by tmilewski** [comment]
!snapshot

**Comment by tmilewski** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @tmilewski - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 1.2.1-snapshot.vfb2ebe7 |
| @clerk/chrome-extension | 1.0.14-snapshot.vfb2ebe7 |
| @clerk/clerk-js | 5.5.1-snapshot.vfb2ebe7 |
| @clerk/elements | 0.5.0-snapshot.vfb2ebe7 |
| @clerk/clerk-expo | 1.1.6-snapshot.vfb2ebe7 |
| @clerk/express | 0.0.10-snapshot.vfb2ebe7 |
| @clerk/fastify | 1.0.11-snapshot.vfb2ebe7 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/nextjs | 5.1.1-snapshot.vfb2ebe7 |
| @clerk/clerk-react | 5.2.1-snapshot.vfb2ebe7 |
| @clerk/remix | 4.0.12-snapshot.vfb2ebe7 |
| @clerk/clerk-sdk-node | 5.0.9-snapshot.vfb2ebe7 |
| @clerk/shared | 2.2.1-snapshot.vfb2ebe7 |
| @clerk/testing | 1.1.4-snapshot.vfb2ebe7 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/backend`
```sh
npm i @clerk/backend@1.2.1-snapshot.vfb2ebe7 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.14-snapshot.vfb2ebe7 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.5.1-snapshot.vfb2ebe7 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.5.0-snapshot.vfb2ebe7 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.6-snapshot.vfb2ebe7 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@0.0.10-snapshot.vfb2ebe7 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.11-snapshot.vfb2ebe7 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.1.1-snapshot.vfb2ebe7 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.2.1-snapshot.vfb2ebe7 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.0.12-snapshot.vfb2ebe7 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.9-snapshot.vfb2ebe7 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.2.1-snapshot.vfb2ebe7 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.1.4-snapshot.vfb2ebe7 --save-exact
```

File: packages/elements/tsup.config.ts [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
 import { defineConfig } from 'tsup';
 
 import { version as clerkJsVersion } from '../clerk-js/package.json';
 import { name, version } from './package.json';
 
+/**
+ * Replaces the inspect with undefined so-as not to bundle
+ * inspectors outside of our development environment.
+ */
+export function dynamicInspectorImport(): Plugin {
```
Comment (tmilewski) [inline packages/elements/tsup.config.ts:11]: Converts exported inspect into `undefined` which is handled appropriately by xstate.

File: packages/elements/tsup.config.ts [hunk 2]
```diff
@@ -21,9 +40,10 @@ export default defineConfig(overrideOptions => {
       'react/sign-in/index': 'src/react/sign-in/index.ts',
       'react/sign-up/index': 'src/react/sign-up/index.ts',
     },
-    external: ['react', 'react-dom'],
+    external: ['react', 'react-dom', '@statelyai/inspect'],
     format: ['cjs', 'esm'],
     minify: false,
     sourcemap: true,
+    esbuildPlugins: isProd ? [dynamicInspectorImport()] : [],
```
[the remark this answers is not in the document]
Comment (LekoArts) [reply packages/elements/tsup.config.ts:47]: +1 to that

File: packages/elements/src/internals/utils/inspector/browser/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { isTruthy } from '@clerk/shared/underscore';
+// @ts-expect-error - Exists only in devDependencies; Removed by tsup for production builds
+// eslint-disable-next-line import/no-unresolved
+import { createBrowserInspector } from '@statelyai/inspect';
+
+export const getInspector = () => {
+  if (
+    typeof window !== 'undefined' &&
+    process.env.NODE_ENV === 'development' &&
+    isTruthy(process.env.NEXT_PUBLIC_CLERK_ELEMENTS_DEBUG_UI)
```
Comment (LekoArts) [inline packages/elements/src/internals/utils/inspector/browser/index.ts:10]: In the `console` file you're also checking for the non-Next.js version, we should do this here, too.
So `CLERK_ELEMENTS_DEBUG_UI`

File: packages/elements/tsup.config.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (BRKalow) [reply packages/elements/tsup.config.ts:47]: Looks like we already have `__DEV__` 👍

[the remark this answers is not in the document]
Comment (tmilewski) [reply packages/elements/tsup.config.ts:47]: Yup, doing that now

**Review by BRKalow — APPROVED** [review]

**Comment by tmilewski** [comment]
Pushing through as integration tests are having issues unrelated to this PR.

**Pull request merged**